src/blocking.rs

```diff
@@ -339,6 +350,28 @@ impl BlockingClient {
         };
         self.get_response_json(&path)
     }
+
+    /// Sends a GET request to the given `url`, retrying failed attempts
+    /// for retryable error codes until max retries hit.
+    pub fn get_with_retry(&self, url: &str) -> Result<Response, Error> {
+        let mut attempts = 0;
+        let mut delay = self.backoff;
+
+        loop {
+            match self.get_request(url)?.send() {
+                Ok(resp)
+                    if attempts < self.max_retries
+                        && RETRYABLE_ERROR_CODES.contains(&(resp.status_code as u16)) =>
```

Comment:
nit: this could be extracted to `is_status_retryable` as we do for `success` and `not_found`